src/utils.ts

```diff
@@ -65,6 +66,148 @@ export const labelClosedPR = async (
   }
 };
 
+const tryBackportAllCommits = async (opts: TryBackportOptions) => {
+  log(
+    'backportImpl',
+    LogLevel.INFO,
+    `Getting rev list from: ${opts.pr.base.sha}..${opts.pr.head.sha}`,
+  );
+
+  const commits = (
+    await opts.context!.github.pulls.listCommits(
+      opts.context!.repo({ pull_number: opts.pr.number }),
+    )
+  ).data.map((commit: PullsListCommitsResponseItem) => commit.sha);
+
+  if (commits.length === 0) {
+    log(
+      'backportImpl',
+      LogLevel.INFO,
+      'Found no commits to backport - aborting backport process',
+    );
+    return false;
+  }
+
+  // Over 240 commits is probably the limit from GitHub so let's not bother.
+  if (commits.length >= 240) {
+    log(
+      'backportImpl',
+      LogLevel.ERROR,
+      `Too many commits (${commits.length})...backport will not be performed.`,
+    );
+    await opts.context!.github.issues.createComment(
+      opts.context!.repo({
+        issue_number: opts.pr.number,
+        body:
+          'This PR has exceeded the automatic backport commit limit \
+and must be performed manually.',
+      }),
+    );
+
+    return false;
+  }
+
+  log(
+    'backportImpl',
+    LogLevel.INFO,
+    `Found ${commits.length} commits to backport - requesting details now.`,
+  );
+
+  const patches: string[] = new Array(commits.length).fill('');
+  const q = makeQueue({ concurrency: 5 });
+  q.stop();
+
+  for (const [i, commit] of commits.entries()) {
+    q.push(async () => {
+      const patchUrl = `https://api.github.com/repos/${opts.slug}/commits/${commit}`;
+      const patchBody = await fetch(patchUrl, {
+        headers: {
+          Accept: 'application/vnd.github.VERSION.patch',
+          Authorization: `token ${opts.repoAccessToken}`,
+        },
+      });
+      patches[i] = await patchBody.text();
+      log(
+        'backportImpl',
+        LogLevel.INFO,
+        `Got patch (${i + 1}/${commits.length})`,
+      );
+    });
+  }
+
+  await new Promise((r) => q.start(r));
```

Comment:
I know this is existing code and I probably wrote it but it's wrong 😆 

```suggestion
  await new Promise((resolve, reject) => q.start(err ? reject(err) : resolve()));
```